PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/storage/component/AccountNameTextField.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+package com.microsoft.azure.toolkit.intellij.storage.component;
+
+import com.azure.core.management.exception.ManagementException;
+import com.microsoft.azure.toolkit.intellij.common.ValidationDebouncedTextInput;
+import com.microsoft.azure.toolkit.lib.Azure;
+import com.microsoft.azure.toolkit.lib.common.entity.CheckNameAvailabilityResultEntity;
+import com.microsoft.azure.toolkit.lib.common.form.AzureValidationInfo;
+import com.microsoft.azure.toolkit.lib.storage.service.AzureStorageAccount;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.function.Function;
+import java.util.regex.Pattern;
+
+public class AccountNameTextField extends ValidationDebouncedTextInput {
+
+    private static final Pattern PATTERN = Pattern.compile("[a-z0-9]{3,24}");
+    @Setter
+    private int minLength = 3;
+    @Setter
+    private int maxLength = 24;
+    @Getter
+    private String subscriptionId;
+    @Setter
+    private Function<AccountNameTextField, AzureValidationInfo> validateFunction;
+
+    public void setSubscriptionId(String subscriptionId) {
+        if (!StringUtils.equals(subscriptionId, this.subscriptionId)) {
+            this.subscriptionId = subscriptionId;
+        }
+    }
+
+    @Override
+    @NotNull
+    public AzureValidationInfo doValidateValue() {
+        if (StringUtils.isBlank(subscriptionId)) {
+            return AzureValidationInfo.UNINITIALIZED;
+        }
+        final AzureValidationInfo info = super.doValidateValue();
+        if (!AzureValidationInfo.OK.equals(info)) {
+            return info;
+        }
+        final String value = this.getValue();
+        // validate length
+        if (StringUtils.length(value) < minLength || StringUtils.length(value) > maxLength) {
+            return AzureValidationInfo.builder().input(this)
+                .message(String.format("Server name must be at least %s characters and at most %s characters.", minLength, maxLength))
+                .type(AzureValidationInfo.Type.ERROR).build();
+        }
+        // validate special character
+        if (!PATTERN.matcher(value).matches()) {
+            return AzureValidationInfo.builder().input(this)
+                .message("The field can contain only lowercase letters and numbers. Name must be between 3 and 24 characters.")
+                .type(AzureValidationInfo.Type.ERROR).build();
+        }
+        // validate availability
+        CheckNameAvailabilityResultEntity resultEntity;
```

Comment:
CheckNameAvailabilityResultEntity -> CheckNameAvailabilityResult

Reply:
It's not belong to the scope of this feature. and it will impact other resources.

and in my opinion, It;s okay.
XXXEntify means a simple POJO, XXX without Entity maybe contain some functions of its object.

Reply:
Result means POJO too.